sktime/transformations/series/bkfilter.py

```diff
@@ -0,0 +1,170 @@
+    that U.S. business cycles typically last from 1.5 to 8 years.
+
+    Parameters
+    ----------
+    low : float
+        Minimum period for oscillations. Baxter and King recommend a value of 6
+        for quarterly data and 1.5 for annual data.
+
+    high : float
+        Maximum period for oscillations. BK recommend 32 for U.S. business cycle
+        quarterly data and 8 for annual data.
+
+    K : int
+        Lead-lag length of the filter. Baxter and King suggest a truncation
+        length of 12 for quarterly data and 3 for annual data.
+
+    Notes
+    -----
+    Returns a centered weighted moving average of the original series. Where
+    the weights a[j] are computed ::
+      a[j] = b[j] + theta, for j = 0, +/-1, +/-2, ... +/- K
+      b[0] = (omega_2 - omega_1)/pi
+      b[j] = 1/(pi*j)(sin(omega_2*j)-sin(omega_1*j), for j = +/-1, +/-2,...
+    and theta is a normalizing constant ::
+      theta = -sum(b)/(2K+1)
+    See the notebook `Time Series Filters
+    <../examples/notebooks/generated/tsa_filters.html>`__ for an overview.
+
+    References
+    ----------
+    Baxter, M. and R. G. King. "Measuring Business Cycles: Approximate
+        Band-Pass Filters for Economic Time Series." *Review of Economics and
+        Statistics*, 1999, 81(4), 575-593.
+
+    Examples
+    --------
+    >>> from sktime.transformations.series.bkfilter import BKFilter
+    >>> import pandas as pd
+    >>> import statsmodels.api as sm
+    >>> dta = sm.datasets.macrodata.load_pandas().data
+    >>> index = pd.date_range(start='1959Q1', end='2009Q4', freq='Q')
+    >>> dta.set_index(index, inplace=True)
+    >>> bk = BKFilter(6, 24, 12)
+    >>> cycles = bk.fit_transform(X=dta[['realinv']])
+    >>> import matplotlib.pyplot as plt
+    >>> fig, ax = plt.subplots()
+    >>> cycles.plot(ax=ax, style=['r--', 'b-'])
+    >>> plt.show()
+    .. plot:: plots/bkf_plot.py
+    """
+
+    _tags = {
+        "scitype:transform-input": "Series",
+        # what is the scitype of X: Series, or Panel
+        "scitype:transform-output": "Series",
+        # what scitype is returned: Primitives, Series, Panel
+        "scitype:instancewise": True,  # is this an instance-wise transform?
+        "univariate-only": False,  # can the transformer handle multivariate X?
+        "X_inner_mtype": "np.ndarray",  # which mtypes do _fit/_predict support for X?
+        # this can be a Panel mtype even if transform-input is Series, vectorized
+        "y_inner_mtype": "None",  # which mtypes do _fit/_predict support for y?
+        "requires_y": False,  # does y need to be passed in fit?
+        "enforce_index_type": [
+            pd.RangeIndex
+        ],  # index type that needs to be enforced in X/y
+        "fit_is_empty": True,  # is fit empty and can be skipped? Yes = True
+        "transform-returns-same-time-index": False,
+        # does transform return have the same time index as input X
+        "capability:unequal_length": False,

```

Comment:
I think this should be `True`, since it is applied per series (so it does not matter if they are different length)

Reply:
Thanks @fkiraly! Totally missed that tag when we did validate the input/output.

@Pyyim, @mgorlin